src/dpop.js

```diff
@@ -0,0 +1,33 @@
+/* @flow */
+
+import { ZalgoPromise } from "@krakenjs/zalgo-promise/src";
+
+export const stringToBytes = (string: string): Uint8Array => {
+  // eslint-disable-next-line compat/compat
+  return new TextEncoder().encode(string);
+};
+
+export const bytesToString = (bytes: Uint8Array): string => {
+  return String.fromCharCode(...bytes);
+};
+
+export const base64encodeUrlSafe = (string: string): string => {
+  // https://datatracker.ietf.org/doc/html/rfc7515#appendix-C
+  return btoa(string)
+    .replace(/[=]+/g, "")
+    .replace(/\+/g, "-")
+    .replace(/\//g, "_");
+};
+
+export const base64decodeUrlSafe = (string: string): string => {
+  return atob(string.replace(/-/g, "+").replace(/_/g, "/"));
+};
+
+export const sha256 = (string: string): ZalgoPromise<string> => {
+  const bytes = stringToBytes(string);
+  return window.crypto.subtle.digest("sha-256", bytes).then((digest) => {
```

Comment:
You could make this a regular promise, since this `.then` would negate any of the synchronous abilities of Zalgo.

Reply:
and if so, i'd suggest going with `async/await`

Reply:
amazing! would you `eslint-disable` these?

```
24:47  error  Unexpected use of 'Promise'  no-restricted-globals
24:47  error  "Promise" is not defined     promise/no-native
```

it looks like it is explicitly enabled here - https://github.com/krakenjs/grumbler-scripts/blame/3d70c3c2a809793b420fadf4c6859016fdcbb094/config/.eslintrc-browser.js#L8-L9 🤔

Reply:
yeah i usually disable them at the top of the file